spring-pulsar-sample-apps/sample-reactive/src/main/java/org.springframework.pulsar.example/ReactiveSpringPulsarBootApp.java

```diff
@@ -50,66 +48,91 @@ public static void main(String[] args) {
 	}
 
 	/**
-	 * Sends POJO messages with a reactive template and receives them with a reactive
-	 * consumer.
+	 * Sends string messages with a reactive template and receives them with a simple
+	 * reactive listener.
 	 */
 	@Configuration(proxyBeanMethods = false)
-	static class ReactiveSenderWithReactiveConsumer implements ApplicationListener<ApplicationReadyEvent> {
+	static class ReactiveTemplateWithSimpleReactiveListener implements ApplicationListener<ApplicationReadyEvent> {
 
 		private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
 		@Autowired
-		private ReactivePulsarTemplate<Foo> reactivePulsarTemplate;
+		private ReactivePulsarTemplate<String> reactivePulsarTemplate;
 
-		// TODO remove this once the auto-config is available
-		@Bean
-		ReactivePulsarConsumerFactory<Foo> reactivePulsarConsumerFactory(ReactivePulsarClient reactivePulsarClient) {
-			MutableReactiveMessageConsumerSpec spec = new MutableReactiveMessageConsumerSpec();
-			spec.setTopicNames(Collections.singletonList("sample-reactive-topic1"));
-			spec.setSubscriptionName("sample-reactive-sub1");
-			spec.setConsumerName("sample-reactive-consumer1");
-			return new DefaultReactivePulsarConsumerFactory<>(reactivePulsarClient, spec);
+		@Override
+		public void onApplicationEvent(ApplicationReadyEvent event) {
+			this.reactivePulsarTemplate
+					.send("sample-reactive-topic1", Flux.range(0, 10).map((i) -> "sample-message-" + i)).subscribe();
```

Comment:
I generally like to use the syntax:
```java
Flux.range(0, 10)
    .map((i) -> "sample-message-" + i)
    .as(messages -> this.reactivePulsarTemplate.send("sample-reactive-topic1", messages))
    .subscribe();
```
WDYT ?

Reply:
Much better. Thx for the tip - my Flux API usage FU is not a well-oiled machine by far.